compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/dfa/FirDataFlowAnalyzer.kt

```diff
@@ -863,6 +865,11 @@ class FirDataFlowAnalyzer(private val components: FirAbstractBodyResolveTransfor
             return variableStorage[symbol]
         }
 
+    // TODO: Fix this -- see broken test_11 in compiler/fir/resolve/testData/resolve/smartcasts/nullability.kt
+    // We currently do not create RealDataFlowVariables for FirQualifiedAccess of symbols of non-local properties.
+    // If we have multiple local variables (could be value parameter) of the same type, there should be separate DataFlowVariables for
+    // accesses to a property for each distinct local variable, i.e., DataFlowVariables in storage for properties should be keyed by
+    // property symbol AND the "owning" local variable. We also need to check that the property is not mutable and has no custom getter.
```

Comment:
Symbol is unique identifier for some declaration (function, class, property, value parameter, etc). They are needed because in current design fir may be changed (I mean that we can create new `FirFunction` instead of old one), but symbol will be saved

So, it's incorrect use some fir as identifier for declaration

Reply:
I'm not proposing to use `FirProperty` or `FirFunction` as identifier (for example)... using the symbol is correct. However, I think we should store the entire "chain" of `FirVariableSymbols` instead of just the last one in the access. In my example:

```
fun foo(a1: A?, a2: A?) {
  a1!!.x!!.y!!
  a2.x.y  
}
```

We should store "not null type info" for the chains `a1`, `a1 -> x`, `a1 -> x -> y`. This way, type info for `a2`, `a2 -> x`, `a2 -> x -> y` are not affected. The current bug is such that the `x` property symbol gets the "not null type info" which affects both `a1` and `a2` in the above example.